spec/unit/rest/Twilio.spec.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+const nock = require('nock');
+var url = require('url');  /* jshint ignore:line */
+
+describe('client', () => {
+  var client;
+  const twilio = require('../../../lib');
+
+  beforeEach(() => {
+    client = new twilio('ACXXXXXXXX', 'test-password');
+  });
+  describe('setting the region', () => {
+    it('should use the default region if only edge is defined', () => {
+      const scope = nock('https://api.edge.us1.twilio.com')
+        .get('/')
+        .reply(200, 'test response');
+      client.edge = 'edge';
+      return client.request({method: 'GET', uri: 'https://api.twilio.com'})
+        .then(() => scope.done());
+    });
+    it('should use the default region if only edge is defined and there is a defined region', () => {
+      const scope = nock('https://api.edge.us1.twilio.com')
+        .get('/')
+        .reply(200, 'test response');
+      client.edge = 'edge';
+      return client.request({method: 'GET', uri: 'https://api.region.twilio.com'})
```

Comment:
Test needed where only `client.region` is set and hostname already contains a region.

Reply:
https://github.com/twilio/twilio-node/pull/563/files#diff-78816a2ef39228b6832a61b7014e7504R62